src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -127,23 +135,33 @@ def get_attrs(
                     ).get("uuid")
             alert_rule_triggers.append(serialized)
 
-        alert_rule_projects = set()
-        for alert_rule in alert_rules.values():
+        alert_activations_ranked = AlertRuleActivations.objects.annotate(
+            rank=Window(
+                expression=RowNumber(),
+                partition_by=[F("alert_rule_id")],
+                order_by=F("date_added").desc(),
+            )
+        )
+        activations = alert_activations_ranked.filter(alert_rule__in=item_list, rank__lte=10)
+        activations_by_alert_rule_id = defaultdict(list)
+        for activation in activations:
+            activations_by_alert_rule_id[activation.alert_rule_id].append(activation)
```

Comment:
Does the order matter here?

Reply:
it _might_
but the client could format the results by date as well? 🤷 

I guess i could sort them here 😅